src/cmake/AxomOptions.cmake

```diff
@@ -13,6 +13,13 @@ if(AXOM_ENABLE_ASAN)
     endif()
 endif()
 
+option(AXOM_ENABLE_UBSAN "Enable UndefinedBehaviorSanitizer for undefined behavior detection (Clang or GCC only)" OFF)
+if(AXOM_ENABLE_UBSAN)
+    if(NOT (C_COMPILER_FAMILY_IS_CLANG OR C_COMPILER_FAMILY_IS_GNU))
+        message(FATAL_ERROR "AXOM_ENABLE_UBSAN only supports Clang and GCC")
+    endif()
+endif()
```

Comment:
Please add a note about the new option to the `RELEASE_NOTES` and to the list of Axom CMake options in the docs